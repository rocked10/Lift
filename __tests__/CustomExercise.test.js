import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { render, } from "@testing-library/react-native";
import CustomExercise from '../screens/customExercise';

describe('Testing Custom Exercise Screen', () => {
    const navigate = jest.fn();
    const wrapper = renderer.create(<CustomExercise navigation={navigate} route={{params: {categoryExercisesSetters: false}}} />);

    it('Should render', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it ("Renders elements", () => {   
        const navigate = jest.fn(); 
        const { getByTestId, getByText, getByPlaceholderText} = render(
                <CustomExercise
                    navigation={navigate}
                    route={{params: {categoryExercisesSetters: false}}}
                />
            );
        getByPlaceholderText('Exercise Name');
        getByPlaceholderText('Youtube Link (Optional)');
        getByTestId('Category Picker');
        getByText('CREATE NEW EXERCISE!')
    });
});