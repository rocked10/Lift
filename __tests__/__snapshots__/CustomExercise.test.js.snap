// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Custom Exercise Screen Should render 1`] = `
<View
  style={
    Object {
      "padding": 16,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontFamily": "lato-bold",
        "fontSize": 22,
        "marginBottom": 16,
      }
    }
  >
    Add a custom exercise! 
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {
          "backgroundColor": "white",
          "borderColor": "#ddd",
          "borderRadius": 6,
          "fontFamily": "lato-regular",
          "margin": 5,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 64,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Exercise Name"
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "zIndex": 1,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "color": "#000000",
              "fontFamily": "lato-regular",
              "fontSize": 18,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {
          "backgroundColor": "white",
          "borderColor": "#ddd",
          "borderRadius": 6,
          "fontFamily": "lato-regular",
          "margin": 5,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 64,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Youtube Link (Optional)"
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "zIndex": 1,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "color": "#000000",
              "fontFamily": "lato-regular",
              "fontSize": 18,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View>
    <RNCPicker
      items={
        Array [
          Object {
            "label": "Olympic",
            "testID": undefined,
            "textColor": undefined,
            "value": "Olympic",
          },
          Object {
            "label": "Legs",
            "testID": undefined,
            "textColor": undefined,
            "value": "Legs",
          },
          Object {
            "label": "Chest",
            "testID": undefined,
            "textColor": undefined,
            "value": "Chest",
          },
          Object {
            "label": "Back",
            "testID": undefined,
            "textColor": undefined,
            "value": "Back",
          },
          Object {
            "label": "Arms",
            "testID": undefined,
            "textColor": undefined,
            "value": "Arms",
          },
          Object {
            "label": "Cardio",
            "testID": undefined,
            "textColor": undefined,
            "value": "Cardio",
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
      testID="Category Picker"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "minWidth": 64,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "karla-bold",
                "fontSize": 16,
              }
            }
          >
            CREATE NEW EXERCISE!
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
